projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-ai.php

```diff
@@ -45,6 +42,38 @@ public function __construct() {
 		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
 	}
 
+	/**
+	 * Checks if a given request is allowed to get AI data from WordPress.com.
+	 *
+	 * @return true|WP_Error True if the request has access, WP_Error object otherwise.
+	 */
+	public function permissions_check() {
+
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			// Only post editors can access the endpoints
+			return new WP_Error(
+				'rest_forbidden',
+				__( 'Sorry, you are not allowed to access Jetpack AI help on this site.', 'jetpack' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			// Only allow WPCOM simple sites and Atomic
+			if ( get_current_site()->id !== 1 ) {
+				return true;
+			}
+		} elseif ( ( new Automattic\Jetpack\Status\Host() )->is_woa_site() ) {
```

Comment:
To ensure backwards compatibility you might prefer to only keep the WoA check on WPCOM context only. This way, when you decide to make the feature available to self-hosted sites, it will also be available in previous versions of the Jetpack plugin. Not a blocker though!